crates/bin/prove_block/src/reexecute.rs

```diff
@@ -147,8 +147,11 @@ pub(crate) fn format_commitment_facts<H: HashFunctionType>(
 impl PerContractStorage for ProverPerContractStorage {
     async fn compute_commitment(&mut self) -> Result<CommitmentInfo, CommitmentInfoError> {
         // TODO: error code
-        let contract_data =
-            self.storage_proof.contract_data.as_ref().expect("storage proof should have a contract_data field");
+        let contract_data = match self.storage_proof.contract_data.as_ref() {
+            None => &ContractData::default(),
+            Some(data) => data,
+        };
```

Comment:
Do `self.storage_proof.contract_data.as_ref().unwrap_or_default()` instead

Reply:
Since we are working with `&ContractData`, it is not straightforward to use methods like `unwrap_or_default` or `unwrap_or`. The variable `contract_data` is expected to be a reference, and implementing `Default` for `&ContractData` directly is not possible because it is a reference, not an owned type. While there are some workarounds to simulate this behavior, the primary goal should be to keep the code simple and readable.